ieeeuoft/hackathon-template
# Added check for different users

## Overview

- This PR identifies participant and admin users (and neither) 
- Displays the correct navbar based on user
- Enforces which endpoints can be viewed by each user
- Added Hardware Site Admins group
- Some refactoring and changing files to typescript
- Fixed some minor errors & code clean up


## Unit Tests Created

- additional tests for withUserCheck (formerly withParticipantCheck)
- additional tests for userSlice (& converted to typescript)
- additional tests for Navbar
- additional tests for Header


## Steps to QA

- Sign in as a participant (someone with a profile), you should not be able to visit the /orders, /teams /reports or /incident-form page
- sign in as an admin (someone with group Hardware Site Admins - run `python manage.py migrate` to apply the migration), you should not be able to visit /cart but all other endpoints are available
- Try to crash Login page
